### ISSUE TITLE
Column collation is missing on foreign keys

### ISSUE THREAD

**Issue by Tobion** [doctrine/dbal#2811]
There are two bugs with column collations:

1. schema update / migrations doesn't recognize collation changes, see #2401 similar issues
2. column collation doesn't work correctly when used on a field referenced as a forgein key.

This tickets is about 2)

When you have a column collation overwritten for an entity primary key that is used as foreign key in another entity, then the collation is MISSING in the foreign key definition. This makes the generated schema invalid because the collations can be different which is not allowed in foreign key constraints.

Example:

```
class Order
{
    /**
     * @var string
     *
     * @ORM\Column(type="string", length=8, options={"fixed"=true, "collation"="ascii_general_ci"})
     * @ORM\Id
     */
    private $id;
```

```
class OrderItem
{
    /**
     * @var Order
     *
     * @ORM\ManyToOne(
     *     targetEntity="AppBundle\Entity\Order",
     *     inversedBy="items"
     * )
     */
    private $order;
```

The generated schema for Order is correct: `id CHAR(8) NOT NULL COLLATE ascii_general_ci`
But OrderItem creates an invalid schema because it is missing the collation of the order primary key: `order_id CHAR(8) NOT NULL` The default collation of the table is used, which can be a totally different one.

When it then tries to add the foreign key constraint it fails with `SQLSTATE[HY000]: General error: 1215 Cannot add foreign key constraint`

**Comment by Ocramius** [comment]
This seems correct: we cannot replicate column information on join columns, since you already went into the dark and unexplored lands of `options`, for which we really can't provide cross-database support.

**Comment by Tobion** [comment]
I don't understand the relation to cross-database support. Join columns just need to apply the same options as the primary key. And if it's hard to do that, then there should at least be an `options` option for the `JoinColumn` annotation. This way one can manually apply the same collation option. But currently there is no such thing which makes the above use-case impossible to achieve.

**Comment by Tobion** [comment]
Btw `options={"fixed"=true,` is also an option. And this get's correctly applies to the foreign key. This is basically the same thing.

**Comment by Tobion** [comment]
Fix provided in https://github.com/doctrine/doctrine2/pull/6830

**Comment by lcobucci** [comment]
@Tobion ORM fix applied and to be released in `v2.6.3`. @Ocramius' concern is mostly due to the fact that we don't guarantee cross-database support on things in `options` - as in: if you decide to change your platform you're on your own :+1:
